framework/src/transfers/MultiAppNearestNodeTransfer.C

```diff
@@ -129,27 +129,14 @@ MultiAppNearestNodeTransfer::execute()
       if (fe_type.order > FIRST && !is_to_nodal)
         mooseError("We don't currently support second order or higher elemental variable ");
 
+      if (!is_to_nodal && isParamValid("target_boundary"))
+        paramWarning("target_boundary",
+                     "Setting a target boundary is only valid for receiving "
+                     "variables of the LAGRANGE basis");
+
       if (is_to_nodal)
       {
-        std::vector<Node *> target_local_nodes;
-
-        if (isParamValid("target_boundary"))
-        {
-          BoundaryID target_bnd_id =
-              _to_meshes[i_to]->getBoundaryID(getParam<BoundaryName>("target_boundary"));
-
-          ConstBndNodeRange & bnd_nodes = *(_to_meshes[i_to])->getBoundaryNodeRange();
-          for (const auto & bnode : bnd_nodes)
-            if (bnode->_bnd_id == target_bnd_id && bnode->_node->processor_id() == processor_id())
-              target_local_nodes.push_back(bnode->_node);
-        }
-        else
-        {
-          target_local_nodes.resize(to_mesh->n_local_nodes());
-          unsigned int i = 0;
-          for (auto & node : to_mesh->local_node_ptr_range())
-            target_local_nodes[i++] = node;
-        }
+        std::vector<Node *> target_local_nodes = getTargetLocalNodes(i_to);
```

Comment:
no need to return to target_local_nodes, getTargetLocalNodes is filling the class attribute _target_local_nodes already

Reply:
I did have a `void` return, but @fdkong preferred returning `_target_local_nodes`.

I'm fine with either - just let me know what you both prefer.

Reply:
if you return _target_local_nodes into a std::vector<Node *> then you are making a copy. 
You can return a reference if you like. Otherwise void is fine to me. Just don't make a copy

Reply:
I also wasn't paying attention and missing returning a reference when `getTargetLocalNodes` is actually called.

Ok, I think this is ready now. Function returns a reference to a member variable both in the function definition and when that function is actually called.